### PR TITLE
CB-8084 conform AppDelegate.m with provisioning profile

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
Hide the two methods in AppDelegate.m for push notifications
when provisioning profile doesn't include the push notifications service
Related to https://issues.apache.org/jira/browse/CB-8084

### CODE REVIEW

File: bin/templates/scripts/cordova/lib/push-notifications.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        exec = require('child_process').exec,
+        PUSH_NOTIFICATIONS_ENABLED_HEADER_FILE = 'PushNotificationsEnabled.h',
+        PUSH_NOTIFICATIONS_ENABLED_HEADER_FILE_CONTENT = '#define PUSH_NOTIFICATIONS_ENABLED ',
+        EXPANDED_PROVISIONING_PROFILE = process.env.EXPANDED_PROVISIONING_PROFILE,
+        PATH_TO_MOBILE_PROVISIONS = path.join(process.env.HOME, 'Library', 'MobileDevice', 'Provisioning Profiles'),
+        PROVISIONING_PROFILE_REQUIRED = process.env.PROVISIONING_PROFILE_REQUIRED,
+        PROVISIONING_PROFILE_UUID_REGEX = new RegExp('<key>UUID<\/key>[\\s\\S]*<string>' + EXPANDED_PROVISIONING_PROFILE + '<\/string>');
+
+    function logErrorIfExists(error) {
+        if (error !== null) {
```
Comment (ligaz) [inline bin/templates/scripts/cordova/lib/push-notifications.js]: You can use just `if(error)` here.

File: bin/templates/scripts/cordova/lib/push-notifications.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    function sanitizeMobileProvision(mobileProvision) {
+        if (mobileProvision.indexOf('.mobileprovision') === -1) {
+            mobileProvision += '.mobileprovision';
+        }
+
+        return mobileProvision;
+    }
+
+    function createPushNotificationsEnabledHeaderFile(pushPluginsEnabled) {
+        var cordovaEnablePluginHeaderFileLocation = path.join(__dirname, '..', '..', process.env.PROJECT_NAME, 'Classes', PUSH_NOTIFICATIONS_ENABLED_HEADER_FILE);
```
Comment (ligaz) [inline bin/templates/scripts/cordova/lib/push-notifications.js]: The dirname of the project file should be available as an environment variable.

File: bin/templates/scripts/cordova/lib/push-notifications.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        fs.writeFileSync(cordovaEnablePluginHeaderFileLocation, PUSH_NOTIFICATIONS_ENABLED_HEADER_FILE_CONTENT + pushPluginsEnabled);
+    }
+
+    function createPushNotificationsEnabledHeaderFileWithMobileProvision(mobileProvision) {
+        var pathToProvisioningProfile = path.join(PATH_TO_MOBILE_PROVISIONS, sanitizeMobileProvision(mobileProvision));
+
+        exec("security cms -D -i \"" + pathToProvisioningProfile + "\"", function (error, stdout, stderr) {
+            logErrorIfExists(error);
+            var pushPluginsEnabled = stdout.indexOf("<key>aps-environment</key>") > -1;
```
Comment (ligaz) [inline bin/templates/scripts/cordova/lib/push-notifications.js]: The variable name should be something like `hasPushNotificationsEntitlement` :smile:

File: bin/templates/scripts/cordova/lib/push-notifications.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            createPushNotificationsEnabledHeaderFile(pushPluginsEnabled);
+        });
+    }
+
+    function isprovisionUUIDSuitable(provisionContents) {
```
Comment (ligaz) [inline bin/templates/scripts/cordova/lib/push-notifications.js]: Maybe the `p` should be capitalized here.

File: bin/templates/scripts/cordova/lib/push-notifications.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        return PROVISIONING_PROFILE_UUID_REGEX.test(provisionContents);
+    }
+
+    function findValidMobileProvision(provisions, successCallback, errorCallback) {
```
Comment (ligaz) [inline bin/templates/scripts/cordova/lib/push-notifications.js]: If you want to follow the node convention the errorCallback should be passed before the success one.

**Comment by Mitko-Kerezov** [comment]
Thanks, @ligaz - comments addressed.
Hey @shazron or @agrieve could someone please take a look at this

**Comment by shazron** [comment]
See the latest comment in https://issues.apache.org/jira/browse/CB-8084 and please continue the discussion there, thanks.

**Comment by Mitko-Kerezov** [comment]
Issue was fixed with https://github.com/apache/cordova-ios/commit/8dcc9dd626c523e6e217dd7e8ce43d0af849fa3a
So I'm closing this now

**Comment by shazron** [comment]
Note that the fix enables it to be disabled using a build.xcconfig preprocessor macro definition (which can be done through a project/plugin hook). In 3.8.0 it still behaves like 3.7.0 by default for backwards compatibility reasons (in 4.0.0, it will be disabled)

**Comment by boboldehampsink** [comment]
Do you have an example for a build hook that implements 

`GCC_PREPROCESSOR_DEFINITIONS = $(inherited) DISABLE_PUSH_NOTIFICATIONS=1`

**Comment by shazron** [comment]
I don't, but you can probably modify this to do what you want: http://stackoverflow.com/a/26563642/219684

**Pull request closed without merging**